### PR TITLE
refactor: avoid DiagnosticResult[] when DiagnosticResult is sufficient.

### PR DESCRIPTION
**Pull request by bcollamore**

### CODE REVIEW

File: Philips.CodeAnalysis.Test/Maintainability/Readability/EnforceRegionsTest.cs [hunk 1]
```diff
@@ -242,45 +242,35 @@ class Foo
...
-				};
-			}
-			else
+			var result = new DiagnosticResult()
 			{
-				results = Array.Empty<DiagnosticResult>();
-			}
-			VerifyDiagnostic(givenText, results);
+				Id = Helper.ToDiagnosticId(DiagnosticId.EnforceNonDuplicateRegion),
+				Message = new System.Text.RegularExpressions.Regex(EnforceRegionsAnalyzer.EnforceNonDuplicateRegionMessageFormat),
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/Maintainability/Readability/EnforceRegionsTest.cs:248]: Add using iso FullyQualifiedName

File: Philips.CodeAnalysis.Test/MsTest/TestClassPublicMethodShouldBeTestMethodTest.cs [hunk 2]
```diff
@@ -108,18 +108,17 @@ private void VerifyError(string baseline, string given, bool isError)
 			string givenText = string.Format(baseline, given);
 			if (isError)
 			{
-				var results = new[] { new DiagnosticResult()
+				var result = new DiagnosticResult()
+				{
+					Id = Helper.ToDiagnosticId(DiagnosticId.TestClassPublicMethodShouldBeTestMethod),
+					Message = new System.Text.RegularExpressions.Regex(TestClassPublicMethodShouldBeTestMethodAnalyzer.MessageFormat),
```
Comment (ynsehoornenborg) [inline Philips.CodeAnalysis.Test/MsTest/TestClassPublicMethodShouldBeTestMethodTest.cs:114]: Using iso FullyQualifiedName

File: Philips.CodeAnalysis.Test/Maintainability/Readability/EnforceRegionsTest.cs (hunk 1 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/Maintainability/Readability/EnforceRegionsTest.cs:248]: done

File: Philips.CodeAnalysis.Test/MsTest/TestClassPublicMethodShouldBeTestMethodTest.cs (hunk 2 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/MsTest/TestClassPublicMethodShouldBeTestMethodTest.cs]: done

File: Philips.CodeAnalysis.Test/Maintainability/Readability/EnforceRegionsTest.cs (hunk 1 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Test/Maintainability/Readability/EnforceRegionsTest.cs:249]: Doesn't show in the diff

**Review by ynsehoornenborg — APPROVED** [review]

File: Philips.CodeAnalysis.Test/Maintainability/Readability/EnforceRegionsTest.cs (hunk 1 above)
Comment (bcollamore) [reply Philips.CodeAnalysis.Test/Maintainability/Readability/EnforceRegionsTest.cs]: done

**Review by ynsehoornenborg — DISMISSED** [review]

**Review by ynsehoornenborg — APPROVED** [review]

**Pull request merged**